src/app/dataset-view/dataset.service.ts

```diff
@@ -226,9 +176,9 @@ export class AppDatasetService {
     ): void {
         this.searchApi
             .getDatasetOverview({ id, page })
-            .subscribe((data: DatasetOverviewQuery | undefined) => {
+            .subscribe((data: {datasets: Datasets} | undefined) => {
                 let datasets: SearchDatasetByID;
-                if (data) {
+                if (!_.isNil(data)) {
                     /* eslint-disable  @typescript-eslint/no-explicit-any */
                     datasets = AppValues.deepCopy(data.datasets.byId);
```

Comment:
I think here we are copying a lot more than we want to copy, including raw data content, which is post-processed below (JSON parsing)